Fe#45/api clients

AwesomeAnders opened the pull request:
Added api clients for a user and a quiz. Some of the backend endpoints doesn't seem to be working correctly. I think some functions are missing e.g. in the addQuiz() method setCreatedBy() is undefined, which results in a internal server error. 

The API Clients are setup so a bearer token is added to the header in each request. In the file web\src\services\api.ts
the authToken should be added and also the hardcoded string for the URL should be changed to match the server-backend-url, but im not quite sure what the correct URL is. Can someone pull and update the value **before merging**? :) 

I've added some TODOs to the quizClient and userClient files. If we wish to parse the JSON-response that we receive from the backend, this should be done here. E.g when a quiz is received from the backend, we might want to parse the JSON object into our own models in the frontend. 

I've added a page on the path '/test' with buttons for each endpoint. You can open the devtools and inspect the network requests if needed. 

If you pull this branch and want to test, remember to change the database URL in the hibernate config file to localhost

EDIT: Also i had a really weird issue where entities are not correctly added to the database. E.g when i POST to /rest/user
with { first_name: bob } an entry is created, but the first_name field is empty... And I have no idea why... I suspect its because of some type-mismatch between the user object in the body of the request and the UserDTO. I need some help from a backend wizard please <3

siffravn commented on src/main/resources/hibernate.cfg.xml:
```diff
@@ -5,9 +5,7 @@
-        <property name="hibernate.connection.username">postgres</property>
-        <property name="hibernate.connection.password">12345</property>
-        <property name="hibernate.connection.url"> jdbc:postgresql://srv-captain--youquiz-db-db:5432/hibernatedb</property>
+        <property name="hibernate.connection.username">postgres</property> <property name="hibernate.connection.password">12345</property> <property name="hibernate.connection.url"> jdbc:postgresql://postgres:5432/hibernatedb</property>
```
These are incorrect conficvalues. Please revert merge from branch BE/dockerCompose

siffravn commented on src/main/java/Util/CORSFilter.java:
```diff
@@ -0,0 +1,20 @@
+package Util;
+
+import java.io.IOException;
+import javax.ws.rs.ext.Provider;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+
+@Provider
+public class CORSFilter implements ContainerResponseFilter {
+
+    @Override
+    public void filter(ContainerRequestContext request, ContainerResponseContext response) throws IOException {
+        response.getHeaders().add("Access-Control-Allow-Origin", "*");
+        response.getHeaders().add("Access-Control-Allow-Headers",
+                "CSRF-Token, X-Requested-By, Authorization, Content-Type");
+        response.getHeaders().add("Access-Control-Allow-Credentials", "true");
+        response.getHeaders().add("Access-Control-Allow-Methods", "GET, POST, PUT, DELETE, OPTIONS, HEAD");
+    }
+}
```
Please resolve conflict with master

AwesomeAnders commented:
@siffravn I reverted merge commit from docker-compose branch and fixed merge conflict with master. 

Before the pull-request is merged the localhost URL in web\src\services\api.ts needs to be changed to the URL of our backend on the server, but I'm not quite sure what the correct URL is. Can you leave a comment here, then i'll update it, or can you pull the branch and fix it? :)
